### PR TITLE
Implement cupy.linalg.inv

### PR DESCRIPTION
**Pull request by rezoo**
This PR implements `cupy.linalg.inv`.

### CODE REVIEW

File: cupy/linalg/solve.py [hunk 1]
```diff
@@ -131,7 +131,26 @@ def tensorsolve(a, b, axes=None):
...
-# TODO(okuta): Implement inv
+def inv(a):
+    '''Computes the inverse of a matrix.
+
+    This function computes matrix ``a_inv`` from n-dimensional regular matrix
+    ``a`` such that ``dot(a, a_inv) == eye(n)``.
+
+    Args:
+        a (cupy.ndarray): The regular matrix
+
```
Comment (okuta) [inline cupy/linalg/solve.py]: Please add `Return` section

File: cupy/linalg/solve.py [hunk 2]
```diff
@@ -131,7 +131,26 @@ def tensorsolve(a, b, axes=None):
...
+    '''
+    if not cuda.cusolver_enabled:
+        raise RuntimeError('Current cupy only supports cusolver in CUDA 8.0')
+
+    util._assert_cupy_array(a)
+    util._assert_rank2(a)
+    util._assert_nd_squareness(a)
+
+    b = cupy.eye(len(a), dtype=a.dtype)
+    return solve(a, b)
```
Comment (okuta) [inline cupy/linalg/solve.py]: Why do not you use [this strategy](https://github.com/chainer/chainer/blob/f552510534ce79dec078f3b11cc724c6a735bf2d/chainer/functions/math/inv.py#L10)?

File: cupy/linalg/solve.py (hunk 1 above)
Comment (rezoo) [reply cupy/linalg/solve.py]: Fixed.

Comment (rezoo) [reply cupy/linalg/solve.py]: Fixed.

File: cupy/linalg/solve.py (hunk 2 above)
Comment (rezoo) [reply cupy/linalg/solve.py]: The above function uses LU decomposition, whereas QR decomposition is used in the implementation of `cupy.linalg.inv`.

In general, it is known that the QR decomposition is numerically stable when solving the linear systems [1]. I think it is better to use QR decomposition in general cases like `cupy.linalg.inv`.
https://www-old.math.gatech.edu/academic/courses/core/math2601/Web-notes/6num.pdf

**Comment by rezoo** [comment]
Thanks, I fixed.

File: cupy/linalg/solve.py (hunk 1 above)
Comment (okuta) [reply cupy/linalg/solve.py]: OK

File: cupy/linalg/solve.py (hunk 2 above)
Comment (okuta) [reply cupy/linalg/solve.py]: OK.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**